arbnode/batch_poster.go

```diff
@@ -553,7 +557,7 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 			b.building = nil
 			return false, fmt.Errorf("error adding message to batch: %w", err)
 		}
-		if !success {
+		if !success && !config.WaitForMaxBatchPostDelay {
 			// this batch is full
 			forcePostBatch = true
 			haveUsefulMessage = true
```

Comment:
We still need to set haveUsefulMessage to true even if this config option is enabled I think

Reply:
done.